wolfcrypt/src/integer.c

```diff
@@ -1313,6 +1302,7 @@ int mp_cmp_d(mp_int * a, mp_digit b)
 void mp_set (mp_int * a, mp_digit b)
 {
   mp_zero (a);
+  mp_grow(a, 1);
```

Comment:
Agreed. I'll investigate effort to add return code checks on mp_set or evaluate another solution.